vm/src/builtins/pytype.rs

```diff
@@ -480,8 +480,10 @@ impl PyType {
         }
 
         // TODO: how do we know if it should have a dict?
```

Comment:
I think this comment is here because the next line is including `HAS_DICT`. Now it is hidden into `py_flags()`. Then the comment also need to be moved.
Otherwise, you seem to find the answer of it. Then we don't need it anymore.

Reply:
Yeah. This isn't the place for this, it should actually be placed exactly above when the `__dict__` is inserted. It's still a `TODO` though because we still don't have slots.

Reply:
Then I think `Self::HAS_DICT` is not belong to `heap_type_flags()`. In that case, I prefer `heap_type_flags() | PyTypeFlags::HAS_DICT` with the comment, to show it is always added for now but that's not because it is the default flag for heap type but we are not implementing type without `__dict__` yet.